### ISSUE TITLE
Inconsistency in complex string representation introduced in NumPy 1.1 (migrated from Trac #841)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2394]
Original ticket http://projects.scipy.org/numpy/ticket/841
Reported 2008-07-04 by atmention:FrancescAlted, assigned to atmention:charris.

With !NumPy 1.0 series, the representation of complex types where identical for !NumPy and Python:

```
>>> numpy.__version__
'1.0.3'
>>> repr(numpy.complex(0))    # The Python type
'0j'
>>> repr(numpy.complex128(0))  # The NumPy type
'0j'
```

Starting with !NumPy 1.1, it diverges from the Python standard:

```
>>> numpy.__version__
'1.2.0.dev5313'
>>> repr(numpy.complex(0))
'0j'
>>> repr(numpy.complex128(0))
'(0.0+0.0j)'
```

I think that, in order to ensure good interactivity of the !NumPy complex type with that of Python, the previous way to represent complex number should be reverted.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2008-07-08

Since no one else on the list seemed to care one way or the other, I'll go ahead and fix this up.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2008-07-12

Fixed in trunk r5395 and backported to 1.1.x.
